src/auth/utils.ts

```diff
@@ -35,14 +36,20 @@ interface AuthTypeApiKey {
   value: string;
 }
 
+interface AuthTypeArtifactKey {
+  type: typeof AuthSecret.ARTIFACT_KEY;
+  value: string;
+}
+
 interface AuthTypeUnknown {
   type: typeof AuthSecret.UNKNOWN;
 }
 
-type AuthType = AuthTypeAccessToken | AuthTypeApiKey | AuthTypeUnknown;
+type AuthType = AuthTypeAccessToken | AuthTypeApiKey | AuthTypeArtifactKey | AuthTypeUnknown;
```

Comment:
I'm not sure that this should be a new Auth type. This secret can not identify the user.

Reply:
It does, not a User resource but it identifies a used of the API (the holder of the shared secret)